### PR TITLE
fix: add a try and catch for delete versioning

### LINKED ISSUE #46737: [Bug]: Error while running background job ExpireVersions
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Error while running background job (class: OCA\\Files_Versions\\BackgroundJob\\ExpireVersion: Exception OCP\\Files\\NotPermittedException
(86 occurrences in 2 days)

### Steps to reproduce

This happens if you have a S3 as primary storage. See background job logs

### Expected behavior

I dont see the error. 

A solution would be:
https://github.com/nextcloud/server/blob/master/apps/files_versions/lib/Storage.php#L596
- to put that line inside a try/catch in order to catch the NotPermittedException, so that it does not exit the loop as soon as an exception like that is thrown
- in the catch, to add an error log message indicating the file id and the exception type

### Installation method

None

### Nextcloud Server version

28

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by GretaD**
* Resolves: #46737 

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_versions/lib/Storage.php [hunk 1]
```diff
@@ -593,8 +594,12 @@ public static function expireOlderThanMaxForUser($uid) {
 				$versionsMapper->delete($versionEntity);
 			}
 
-			$version->delete();
-			\OC_Hook::emit('\OCP\Versions', 'delete', ['path' => $internalPath, 'trigger' => self::DELETE_TRIGGER_RETENTION_CONSTRAINT]);
+			try {
+				$version->delete();
+				\OC_Hook::emit('\OCP\Versions', 'delete', ['path' => $internalPath, 'trigger' => self::DELETE_TRIGGER_RETENTION_CONSTRAINT]);
+			} catch (NotPermittedException $e) {
+				\OC::$server->get(LoggerInterface::class)->error("Error deleting version: File ID {$internalPath}");
```
Comment (artonge) [inline apps/files_versions/lib/Storage.php:601]:
```suggestion
				\OCP\Server::get(LoggerInterface::class)->error("Missing permissions to delete version: {$internalPath}", ['app' => 'files_versions', 'exception' => $e]);
```

**Review by artonge — APPROVED** [review]
Looks good. Tested and it works. :)

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**